src/scripts/main.js

```diff
@@ -1 +1,65 @@
 'use strict';
+
+const doc = document.querySelector('body');
+
+const firstPromise = new Promise((resolve, reject) => {
+  doc.addEventListener('click', () => {
+    resolve();
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  doc.addEventListener('click', () => {
+    resolve();
+  });
+
+  doc.addEventListener('contextmenu', () => {
+    resolve();
+  });
```

Comment:
The `secondPromise` resolves on either a click or a context menu event. Consider handling the case where both events occur to ensure consistent behavior with `thirdPromise`.